[2025-07-01T11:36:58.954+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T11:36:58.970+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T11:36:58.974+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T11:36:58.975+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-07-01T11:36:58.981+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T11:36:58.985+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpfkgzgvea']
[2025-07-01T11:36:58.986+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask submit_grazing_detector_job
[2025-07-01T11:36:58.986+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=240) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T11:36:58.986+0000] {standard_task_runner.py:63} INFO - Started process 242 to run task
[2025-07-01T11:36:59.013+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host db0f7c55fcb6
[2025-07-01T11:36:59.052+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T11:36:59.052+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T11:36:59.075+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-07-01T11:36:59.075+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master yarn --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark --queue root.default /opt/***/spark_jobs/grazing_detector.py kafka:29092
[2025-07-01T11:36:59.114+0000] {spark_submit.py:491} INFO - JAVA_HOME is not set
[2025-07-01T11:36:59.114+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T11:36:59.119+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark --queue root.default /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.
[2025-07-01T11:36:59.121+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T113658, end_date=20250701T113659
[2025-07-01T11:36:59.126+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task submit_grazing_detector_job (Cannot execute: spark-submit --master yarn --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark --queue root.default /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.; 242)
[2025-07-01T11:36:59.167+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T11:36:59.175+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T11:36:59.176+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-01T12:10:21.953+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T12:10:21.967+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:10:21.972+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:10:21.973+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-07-01T12:10:21.980+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T12:10:21.984+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpuu3w5d6a']
[2025-07-01T12:10:21.987+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=181) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T12:10:21.987+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask submit_grazing_detector_job
[2025-07-01T12:10:21.987+0000] {standard_task_runner.py:63} INFO - Started process 183 to run task
[2025-07-01T12:10:22.014+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host 3ceedc259e60
[2025-07-01T12:10:22.051+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T12:10:22.051+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T12:10:22.071+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-07-01T12:10:22.072+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092
[2025-07-01T12:10:22.115+0000] {spark_submit.py:491} INFO - JAVA_HOME is not set
[2025-07-01T12:10:22.115+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T12:10:22.120+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.
[2025-07-01T12:10:22.123+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T121021, end_date=20250701T121022
[2025-07-01T12:10:22.128+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task submit_grazing_detector_job (Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.; 183)
[2025-07-01T12:10:22.167+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T12:10:22.178+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T12:10:22.179+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-01T12:13:42.487+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T12:13:42.506+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:13:42.513+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:13:42.513+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-07-01T12:13:42.520+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T12:13:42.524+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpvh946mde']
[2025-07-01T12:13:42.525+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=251) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T12:13:42.525+0000] {standard_task_runner.py:91} INFO - Job 6: Subtask submit_grazing_detector_job
[2025-07-01T12:13:42.525+0000] {standard_task_runner.py:63} INFO - Started process 252 to run task
[2025-07-01T12:13:42.552+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host 3ceedc259e60
[2025-07-01T12:13:42.588+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T12:13:42.589+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T12:13:42.598+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-07-01T12:13:42.599+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092
[2025-07-01T12:13:42.632+0000] {spark_submit.py:491} INFO - JAVA_HOME is not set
[2025-07-01T12:13:42.633+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T12:13:42.638+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.
[2025-07-01T12:13:42.639+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T121342, end_date=20250701T121342
[2025-07-01T12:13:42.643+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 6 for task submit_grazing_detector_job (Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.; 252)
[2025-07-01T12:13:42.668+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T12:13:42.676+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T12:13:42.677+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-01T12:23:00.695+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T12:23:00.710+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:23:00.715+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:23:00.716+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-07-01T12:23:00.722+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T12:23:00.727+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpb5qh9rs6']
[2025-07-01T12:23:00.728+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=181) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T12:23:00.728+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask submit_grazing_detector_job
[2025-07-01T12:23:00.728+0000] {standard_task_runner.py:63} INFO - Started process 183 to run task
[2025-07-01T12:23:00.753+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host 34354cb0d819
[2025-07-01T12:23:00.787+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T12:23:00.787+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T12:23:00.808+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-07-01T12:23:00.809+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092
[2025-07-01T12:23:00.850+0000] {spark_submit.py:491} INFO - JAVA_HOME is not set
[2025-07-01T12:23:00.851+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T12:23:00.856+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.
[2025-07-01T12:23:00.858+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T122300, end_date=20250701T122300
[2025-07-01T12:23:00.862+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task submit_grazing_detector_job (Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.; 183)
[2025-07-01T12:23:00.868+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T12:23:00.877+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T12:23:00.878+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-01T12:28:53.778+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T12:28:53.793+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:28:53.796+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:28:53.796+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-07-01T12:28:53.800+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T12:28:53.802+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpop7fumh9']
[2025-07-01T12:28:53.804+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask submit_grazing_detector_job
[2025-07-01T12:28:53.803+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=180) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T12:28:53.804+0000] {standard_task_runner.py:63} INFO - Started process 181 to run task
[2025-07-01T12:28:53.825+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host 517ac1708c86
[2025-07-01T12:28:53.861+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T12:28:53.861+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T12:28:53.883+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-07-01T12:28:53.883+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092
[2025-07-01T12:28:53.925+0000] {spark_submit.py:491} INFO - JAVA_HOME is not set
[2025-07-01T12:28:53.925+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T12:28:53.930+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.
[2025-07-01T12:28:53.932+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T122853, end_date=20250701T122853
[2025-07-01T12:28:53.936+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task submit_grazing_detector_job (Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.; 181)
[2025-07-01T12:28:53.946+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T12:28:53.954+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T12:28:53.954+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-01T12:37:53.331+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T12:37:53.348+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:37:53.351+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T12:37:53.351+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-07-01T12:37:53.357+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T12:37:53.361+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpew6zi12d']
[2025-07-01T12:37:53.363+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=174) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T12:37:53.363+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask submit_grazing_detector_job
[2025-07-01T12:37:53.363+0000] {standard_task_runner.py:63} INFO - Started process 175 to run task
[2025-07-01T12:37:53.390+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host 9ee5ad7012fb
[2025-07-01T12:37:53.435+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T12:37:53.436+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T12:37:53.462+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-07-01T12:37:53.463+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092
[2025-07-01T12:37:53.505+0000] {spark_submit.py:491} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/spark-class: line 71: /usr/lib/jvm/java-17-openjdk-amd64/bin/java: No such file or directory
[2025-07-01T12:37:53.506+0000] {spark_submit.py:491} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/spark-class: line 97: CMD: bad array subscript
[2025-07-01T12:37:53.508+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T12:37:53.514+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.
[2025-07-01T12:37:53.516+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T123753, end_date=20250701T123753
[2025-07-01T12:37:53.521+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task submit_grazing_detector_job (Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.; 175)
[2025-07-01T12:37:53.544+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T12:37:53.552+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T12:37:53.553+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-01T13:03:30.714+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T13:03:30.731+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T13:03:30.735+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T13:03:30.735+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-07-01T13:03:30.741+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T13:03:30.744+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpgm6rt6n1']
[2025-07-01T13:03:30.746+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=180) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T13:03:30.746+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask submit_grazing_detector_job
[2025-07-01T13:03:30.746+0000] {standard_task_runner.py:63} INFO - Started process 181 to run task
[2025-07-01T13:03:30.781+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host 2b2f1149d0f1
[2025-07-01T13:03:30.823+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T13:03:30.824+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T13:03:30.847+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-07-01T13:03:30.848+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092
[2025-07-01T13:03:31.570+0000] {spark_submit.py:491} INFO - :: loading settings :: url = jar:file:/opt/spark-3.5.1-bin-hadoop3/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-07-01T13:03:31.607+0000] {spark_submit.py:491} INFO - Ivy Default Cache set to: /home/***/.ivy2/cache
[2025-07-01T13:03:31.608+0000] {spark_submit.py:491} INFO - The jars for the packages stored in: /home/***/.ivy2/jars
[2025-07-01T13:03:31.609+0000] {spark_submit.py:491} INFO - org.apache.spark#spark-sql-kafka-0-10_2.13 added as a dependency
[2025-07-01T13:03:31.610+0000] {spark_submit.py:491} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-fcd3a35c-dfb0-452a-b8a7-97a890b58b95;1.0
[2025-07-01T13:03:31.610+0000] {spark_submit.py:491} INFO - confs: [default]
[2025-07-01T13:03:35.299+0000] {spark_submit.py:491} INFO - found org.apache.spark#spark-sql-kafka-0-10_2.13;3.5.1 in central
[2025-07-01T13:03:35.755+0000] {spark_submit.py:491} INFO - found org.apache.spark#spark-token-provider-kafka-0-10_2.13;3.5.1 in central
[2025-07-01T13:03:36.102+0000] {spark_submit.py:491} INFO - found org.apache.kafka#kafka-clients;3.4.1 in central
[2025-07-01T13:03:36.450+0000] {spark_submit.py:491} INFO - found org.lz4#lz4-java;1.8.0 in central
[2025-07-01T13:03:36.805+0000] {spark_submit.py:491} INFO - found org.xerial.snappy#snappy-java;1.1.10.3 in central
[2025-07-01T13:03:38.147+0000] {spark_submit.py:491} INFO - found org.slf4j#slf4j-api;2.0.7 in central
[2025-07-01T13:03:40.515+0000] {spark_submit.py:491} INFO - found org.apache.hadoop#hadoop-client-runtime;3.3.4 in central
[2025-07-01T13:03:40.873+0000] {spark_submit.py:491} INFO - found org.apache.hadoop#hadoop-client-api;3.3.4 in central
[2025-07-01T13:03:43.261+0000] {spark_submit.py:491} INFO - found commons-logging#commons-logging;1.1.3 in central
[2025-07-01T13:03:43.597+0000] {spark_submit.py:491} INFO - found com.google.code.findbugs#jsr305;3.0.0 in central
[2025-07-01T13:03:43.938+0000] {spark_submit.py:491} INFO - found org.scala-lang.modules#scala-parallel-collections_2.13;1.0.4 in central
[2025-07-01T13:03:47.159+0000] {spark_submit.py:491} INFO - found org.apache.commons#commons-pool2;2.11.1 in central
[2025-07-01T13:03:47.327+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/apache/spark/spark-sql-kafka-0-10_2.13/3.5.1/spark-sql-kafka-0-10_2.13-3.5.1.jar ...
[2025-07-01T13:03:47.723+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.apache.spark#spark-sql-kafka-0-10_2.13;3.5.1!spark-sql-kafka-0-10_2.13.jar (559ms)
[2025-07-01T13:03:47.888+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/apache/spark/spark-token-provider-kafka-0-10_2.13/3.5.1/spark-token-provider-kafka-0-10_2.13-3.5.1.jar ...
[2025-07-01T13:03:48.063+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.apache.spark#spark-token-provider-kafka-0-10_2.13;3.5.1!spark-token-provider-kafka-0-10_2.13.jar (338ms)
[2025-07-01T13:03:48.232+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/scala-lang/modules/scala-parallel-collections_2.13/1.0.4/scala-parallel-collections_2.13-1.0.4.jar ...
[2025-07-01T13:03:48.607+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.scala-lang.modules#scala-parallel-collections_2.13;1.0.4!scala-parallel-collections_2.13.jar (544ms)
[2025-07-01T13:03:48.777+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/apache/kafka/kafka-clients/3.4.1/kafka-clients-3.4.1.jar ...
[2025-07-01T13:03:49.280+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.apache.kafka#kafka-clients;3.4.1!kafka-clients.jar (672ms)
[2025-07-01T13:03:49.444+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/com/google/code/findbugs/jsr305/3.0.0/jsr305-3.0.0.jar ...
[2025-07-01T13:03:49.613+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] com.google.code.findbugs#jsr305;3.0.0!jsr305.jar (332ms)
[2025-07-01T13:03:49.776+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/apache/commons/commons-pool2/2.11.1/commons-pool2-2.11.1.jar ...
[2025-07-01T13:03:49.945+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.apache.commons#commons-pool2;2.11.1!commons-pool2.jar (330ms)
[2025-07-01T13:03:50.110+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-runtime/3.3.4/hadoop-client-runtime-3.3.4.jar ...
[2025-07-01T13:03:51.621+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.apache.hadoop#hadoop-client-runtime;3.3.4!hadoop-client-runtime.jar (1675ms)
[2025-07-01T13:03:51.797+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/lz4/lz4-java/1.8.0/lz4-java-1.8.0.jar ...
[2025-07-01T13:03:51.979+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.lz4#lz4-java;1.8.0!lz4-java.jar (357ms)
[2025-07-01T13:03:52.142+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/xerial/snappy/snappy-java/1.1.10.3/snappy-java-1.1.10.3.jar ...
[2025-07-01T13:03:52.358+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.xerial.snappy#snappy-java;1.1.10.3!snappy-java.jar(bundle) (378ms)
[2025-07-01T13:03:52.520+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/slf4j/slf4j-api/2.0.7/slf4j-api-2.0.7.jar ...
[2025-07-01T13:03:52.687+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.slf4j#slf4j-api;2.0.7!slf4j-api.jar (328ms)
[2025-07-01T13:03:52.849+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-api/3.3.4/hadoop-client-api-3.3.4.jar ...
[2025-07-01T13:03:53.744+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] org.apache.hadoop#hadoop-client-api;3.3.4!hadoop-client-api.jar (1057ms)
[2025-07-01T13:03:53.907+0000] {spark_submit.py:491} INFO - downloading https://repo1.maven.org/maven2/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar ...
[2025-07-01T13:03:54.076+0000] {spark_submit.py:491} INFO - [SUCCESSFUL ] commons-logging#commons-logging;1.1.3!commons-logging.jar (331ms)
[2025-07-01T13:03:54.077+0000] {spark_submit.py:491} INFO - :: resolution report :: resolve 15554ms :: artifacts dl 6912ms
[2025-07-01T13:03:54.078+0000] {spark_submit.py:491} INFO - :: modules in use:
[2025-07-01T13:03:54.078+0000] {spark_submit.py:491} INFO - com.google.code.findbugs#jsr305;3.0.0 from central in [default]
[2025-07-01T13:03:54.079+0000] {spark_submit.py:491} INFO - commons-logging#commons-logging;1.1.3 from central in [default]
[2025-07-01T13:03:54.080+0000] {spark_submit.py:491} INFO - org.apache.commons#commons-pool2;2.11.1 from central in [default]
[2025-07-01T13:03:54.080+0000] {spark_submit.py:491} INFO - org.apache.hadoop#hadoop-client-api;3.3.4 from central in [default]
[2025-07-01T13:03:54.081+0000] {spark_submit.py:491} INFO - org.apache.hadoop#hadoop-client-runtime;3.3.4 from central in [default]
[2025-07-01T13:03:54.081+0000] {spark_submit.py:491} INFO - org.apache.kafka#kafka-clients;3.4.1 from central in [default]
[2025-07-01T13:03:54.082+0000] {spark_submit.py:491} INFO - org.apache.spark#spark-sql-kafka-0-10_2.13;3.5.1 from central in [default]
[2025-07-01T13:03:54.082+0000] {spark_submit.py:491} INFO - org.apache.spark#spark-token-provider-kafka-0-10_2.13;3.5.1 from central in [default]
[2025-07-01T13:03:54.082+0000] {spark_submit.py:491} INFO - org.lz4#lz4-java;1.8.0 from central in [default]
[2025-07-01T13:03:54.082+0000] {spark_submit.py:491} INFO - org.scala-lang.modules#scala-parallel-collections_2.13;1.0.4 from central in [default]
[2025-07-01T13:03:54.083+0000] {spark_submit.py:491} INFO - org.slf4j#slf4j-api;2.0.7 from central in [default]
[2025-07-01T13:03:54.083+0000] {spark_submit.py:491} INFO - org.xerial.snappy#snappy-java;1.1.10.3 from central in [default]
[2025-07-01T13:03:54.083+0000] {spark_submit.py:491} INFO - ---------------------------------------------------------------------
[2025-07-01T13:03:54.084+0000] {spark_submit.py:491} INFO - |                  |            modules            ||   artifacts   |
[2025-07-01T13:03:54.084+0000] {spark_submit.py:491} INFO - |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-07-01T13:03:54.084+0000] {spark_submit.py:491} INFO - ---------------------------------------------------------------------
[2025-07-01T13:03:54.084+0000] {spark_submit.py:491} INFO - |      default     |   12  |   12  |   12  |   0   ||   12  |   12  |
[2025-07-01T13:03:54.084+0000] {spark_submit.py:491} INFO - ---------------------------------------------------------------------
[2025-07-01T13:03:54.085+0000] {spark_submit.py:491} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-fcd3a35c-dfb0-452a-b8a7-97a890b58b95
[2025-07-01T13:03:54.085+0000] {spark_submit.py:491} INFO - confs: [default]
[2025-07-01T13:03:54.137+0000] {spark_submit.py:491} INFO - 12 artifacts copied, 0 already retrieved (57876kB/53ms)
[2025-07-01T13:03:54.226+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-07-01T13:03:54.838+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SparkContext: Running Spark version 3.5.1
[2025-07-01T13:03:54.839+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SparkContext: OS info Linux, 6.10.14-linuxkit, aarch64
[2025-07-01T13:03:54.839+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SparkContext: Java version 17.0.15
[2025-07-01T13:03:54.845+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO ResourceUtils: ==============================================================
[2025-07-01T13:03:54.846+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-07-01T13:03:54.846+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO ResourceUtils: ==============================================================
[2025-07-01T13:03:54.846+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SparkContext: Submitted application: GrazingDetectorPy
[2025-07-01T13:03:54.851+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-07-01T13:03:54.854+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO ResourceProfile: Limiting resource is cpu
[2025-07-01T13:03:54.854+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-07-01T13:03:54.872+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SecurityManager: Changing view acls to: ***
[2025-07-01T13:03:54.872+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SecurityManager: Changing modify acls to: ***
[2025-07-01T13:03:54.872+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SecurityManager: Changing view acls groups to:
[2025-07-01T13:03:54.872+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SecurityManager: Changing modify acls groups to:
[2025-07-01T13:03:54.873+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-07-01T13:03:54.952+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO Utils: Successfully started service 'sparkDriver' on port 35937.
[2025-07-01T13:03:54.961+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SparkEnv: Registering MapOutputTracker
[2025-07-01T13:03:54.975+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SparkEnv: Registering BlockManagerMaster
[2025-07-01T13:03:54.982+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-07-01T13:03:54.982+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-07-01T13:03:54.983+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-07-01T13:03:54.991+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-5830f7d6-3b99-43b2-b394-bb3ed5ec0d40
[2025-07-01T13:03:54.996+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:54 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-07-01T13:03:55.002+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-07-01T13:03:55.048+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-07-01T13:03:55.076+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-07-01T13:03:55.090+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.apache.spark_spark-sql-kafka-0-10_2.13-3.5.1.jar at spark://2b2f1149d0f1:35937/jars/org.apache.spark_spark-sql-kafka-0-10_2.13-3.5.1.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.apache.spark_spark-token-provider-kafka-0-10_2.13-3.5.1.jar at spark://2b2f1149d0f1:35937/jars/org.apache.spark_spark-token-provider-kafka-0-10_2.13-3.5.1.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.scala-lang.modules_scala-parallel-collections_2.13-1.0.4.jar at spark://2b2f1149d0f1:35937/jars/org.scala-lang.modules_scala-parallel-collections_2.13-1.0.4.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.apache.kafka_kafka-clients-3.4.1.jar at spark://2b2f1149d0f1:35937/jars/org.apache.kafka_kafka-clients-3.4.1.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/com.google.code.findbugs_jsr305-3.0.0.jar at spark://2b2f1149d0f1:35937/jars/com.google.code.findbugs_jsr305-3.0.0.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.apache.commons_commons-pool2-2.11.1.jar at spark://2b2f1149d0f1:35937/jars/org.apache.commons_commons-pool2-2.11.1.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.apache.hadoop_hadoop-client-runtime-3.3.4.jar at spark://2b2f1149d0f1:35937/jars/org.apache.hadoop_hadoop-client-runtime-3.3.4.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.lz4_lz4-java-1.8.0.jar at spark://2b2f1149d0f1:35937/jars/org.lz4_lz4-java-1.8.0.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.xerial.snappy_snappy-java-1.1.10.3.jar at spark://2b2f1149d0f1:35937/jars/org.xerial.snappy_snappy-java-1.1.10.3.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.slf4j_slf4j-api-2.0.7.jar at spark://2b2f1149d0f1:35937/jars/org.slf4j_slf4j-api-2.0.7.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/org.apache.hadoop_hadoop-client-api-3.3.4.jar at spark://2b2f1149d0f1:35937/jars/org.apache.hadoop_hadoop-client-api-3.3.4.jar with timestamp 1751375034835
[2025-07-01T13:03:55.091+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added JAR file:///home/***/.ivy2/jars/commons-logging_commons-logging-1.1.3.jar at spark://2b2f1149d0f1:35937/jars/commons-logging_commons-logging-1.1.3.jar with timestamp 1751375034835
[2025-07-01T13:03:55.092+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.apache.spark_spark-sql-kafka-0-10_2.13-3.5.1.jar at spark://2b2f1149d0f1:35937/files/org.apache.spark_spark-sql-kafka-0-10_2.13-3.5.1.jar with timestamp 1751375034835
[2025-07-01T13:03:55.093+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.apache.spark_spark-sql-kafka-0-10_2.13-3.5.1.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.apache.spark_spark-sql-kafka-0-10_2.13-3.5.1.jar
[2025-07-01T13:03:55.098+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.apache.spark_spark-token-provider-kafka-0-10_2.13-3.5.1.jar at spark://2b2f1149d0f1:35937/files/org.apache.spark_spark-token-provider-kafka-0-10_2.13-3.5.1.jar with timestamp 1751375034835
[2025-07-01T13:03:55.098+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.apache.spark_spark-token-provider-kafka-0-10_2.13-3.5.1.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.apache.spark_spark-token-provider-kafka-0-10_2.13-3.5.1.jar
[2025-07-01T13:03:55.100+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.scala-lang.modules_scala-parallel-collections_2.13-1.0.4.jar at spark://2b2f1149d0f1:35937/files/org.scala-lang.modules_scala-parallel-collections_2.13-1.0.4.jar with timestamp 1751375034835
[2025-07-01T13:03:55.100+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.scala-lang.modules_scala-parallel-collections_2.13-1.0.4.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.scala-lang.modules_scala-parallel-collections_2.13-1.0.4.jar
[2025-07-01T13:03:55.102+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.apache.kafka_kafka-clients-3.4.1.jar at spark://2b2f1149d0f1:35937/files/org.apache.kafka_kafka-clients-3.4.1.jar with timestamp 1751375034835
[2025-07-01T13:03:55.102+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.apache.kafka_kafka-clients-3.4.1.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.apache.kafka_kafka-clients-3.4.1.jar
[2025-07-01T13:03:55.106+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/com.google.code.findbugs_jsr305-3.0.0.jar at spark://2b2f1149d0f1:35937/files/com.google.code.findbugs_jsr305-3.0.0.jar with timestamp 1751375034835
[2025-07-01T13:03:55.106+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/com.google.code.findbugs_jsr305-3.0.0.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/com.google.code.findbugs_jsr305-3.0.0.jar
[2025-07-01T13:03:55.108+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.apache.commons_commons-pool2-2.11.1.jar at spark://2b2f1149d0f1:35937/files/org.apache.commons_commons-pool2-2.11.1.jar with timestamp 1751375034835
[2025-07-01T13:03:55.108+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.apache.commons_commons-pool2-2.11.1.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.apache.commons_commons-pool2-2.11.1.jar
[2025-07-01T13:03:55.109+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.apache.hadoop_hadoop-client-runtime-3.3.4.jar at spark://2b2f1149d0f1:35937/files/org.apache.hadoop_hadoop-client-runtime-3.3.4.jar with timestamp 1751375034835
[2025-07-01T13:03:55.110+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.apache.hadoop_hadoop-client-runtime-3.3.4.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.apache.hadoop_hadoop-client-runtime-3.3.4.jar
[2025-07-01T13:03:55.125+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.lz4_lz4-java-1.8.0.jar at spark://2b2f1149d0f1:35937/files/org.lz4_lz4-java-1.8.0.jar with timestamp 1751375034835
[2025-07-01T13:03:55.125+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.lz4_lz4-java-1.8.0.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.lz4_lz4-java-1.8.0.jar
[2025-07-01T13:03:55.127+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.xerial.snappy_snappy-java-1.1.10.3.jar at spark://2b2f1149d0f1:35937/files/org.xerial.snappy_snappy-java-1.1.10.3.jar with timestamp 1751375034835
[2025-07-01T13:03:55.127+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.xerial.snappy_snappy-java-1.1.10.3.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.xerial.snappy_snappy-java-1.1.10.3.jar
[2025-07-01T13:03:55.131+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.slf4j_slf4j-api-2.0.7.jar at spark://2b2f1149d0f1:35937/files/org.slf4j_slf4j-api-2.0.7.jar with timestamp 1751375034835
[2025-07-01T13:03:55.131+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.slf4j_slf4j-api-2.0.7.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.slf4j_slf4j-api-2.0.7.jar
[2025-07-01T13:03:55.133+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/org.apache.hadoop_hadoop-client-api-3.3.4.jar at spark://2b2f1149d0f1:35937/files/org.apache.hadoop_hadoop-client-api-3.3.4.jar with timestamp 1751375034835
[2025-07-01T13:03:55.133+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/org.apache.hadoop_hadoop-client-api-3.3.4.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/org.apache.hadoop_hadoop-client-api-3.3.4.jar
[2025-07-01T13:03:55.143+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Added file file:///home/***/.ivy2/jars/commons-logging_commons-logging-1.1.3.jar at spark://2b2f1149d0f1:35937/files/commons-logging_commons-logging-1.1.3.jar with timestamp 1751375034835
[2025-07-01T13:03:55.144+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Copying /home/***/.ivy2/jars/commons-logging_commons-logging-1.1.3.jar to /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/userFiles-05d03066-7913-4913-89e1-7c95ae0e9664/commons-logging_commons-logging-1.1.3.jar
[2025-07-01T13:03:55.177+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-07-01T13:03:55.195+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO TransportClientFactory: Successfully created connection to spark-master/172.18.0.3:7077 after 10 ms (0 ms spent in bootstraps)
[2025-07-01T13:03:55.260+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250701130355-0000
[2025-07-01T13:03:55.264+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 37825.
[2025-07-01T13:03:55.264+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO NettyBlockTransferService: Server created on 2b2f1149d0f1:37825
[2025-07-01T13:03:55.265+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-07-01T13:03:55.267+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250701130355-0000/0 on worker-20250701130225-172.18.0.7-42781 (172.18.0.7:42781) with 14 core(s)
[2025-07-01T13:03:55.268+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO StandaloneSchedulerBackend: Granted executor ID app-20250701130355-0000/0 on hostPort 172.18.0.7:42781 with 14 core(s), 1024.0 MiB RAM
[2025-07-01T13:03:55.268+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2b2f1149d0f1, 37825, None)
[2025-07-01T13:03:55.270+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO BlockManagerMasterEndpoint: Registering block manager 2b2f1149d0f1:37825 with 434.4 MiB RAM, BlockManagerId(driver, 2b2f1149d0f1, 37825, None)
[2025-07-01T13:03:55.271+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2b2f1149d0f1, 37825, None)
[2025-07-01T13:03:55.271+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2b2f1149d0f1, 37825, None)
[2025-07-01T13:03:55.355+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250701130355-0000/0 is now RUNNING
[2025-07-01T13:03:55.364+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-07-01T13:03:55.475+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-07-01T13:03:55.476+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-07-01T13:03:55.836+0000] {spark_submit.py:491} INFO - Traceback (most recent call last):
[2025-07-01T13:03:55.837+0000] {spark_submit.py:491} INFO - File "/opt/***/spark_jobs/grazing_detector.py", line 127, in <module>
[2025-07-01T13:03:55.837+0000] {spark_submit.py:491} INFO - main()
[2025-07-01T13:03:55.837+0000] {spark_submit.py:491} INFO - File "/opt/***/spark_jobs/grazing_detector.py", line 72, in main
[2025-07-01T13:03:55.837+0000] {spark_submit.py:491} INFO - .load()
[2025-07-01T13:03:55.838+0000] {spark_submit.py:491} INFO - ^^^^^^
[2025-07-01T13:03:55.838+0000] {spark_submit.py:491} INFO - File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/streaming/readwriter.py", line 304, in load
[2025-07-01T13:03:55.838+0000] {spark_submit.py:491} INFO - File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-07-01T13:03:55.838+0000] {spark_submit.py:491} INFO - File "/opt/spark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2025-07-01T13:03:55.838+0000] {spark_submit.py:491} INFO - File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-07-01T13:03:55.841+0000] {spark_submit.py:491} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o32.load.
[2025-07-01T13:03:55.845+0000] {spark_submit.py:491} INFO - : java.lang.NoClassDefFoundError: scala/$less$colon$less
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at org.apache.spark.sql.kafka010.KafkaSourceProvider.org$apache$spark$sql$kafka010$KafkaSourceProvider$$validateStreamOptions(KafkaSourceProvider.scala:338)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at org.apache.spark.sql.kafka010.KafkaSourceProvider.sourceSchema(KafkaSourceProvider.scala:71)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at org.apache.spark.sql.execution.datasources.DataSource.sourceSchema(DataSource.scala:233)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at org.apache.spark.sql.execution.datasources.DataSource.sourceInfo$lzycompute(DataSource.scala:118)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at org.apache.spark.sql.execution.datasources.DataSource.sourceInfo(DataSource.scala:118)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at org.apache.spark.sql.execution.streaming.StreamingRelation$.apply(StreamingRelation.scala:36)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at org.apache.spark.sql.streaming.DataStreamReader.loadInternal(DataStreamReader.scala:169)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at org.apache.spark.sql.streaming.DataStreamReader.load(DataStreamReader.scala:145)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-07-01T13:03:55.846+0000] {spark_submit.py:491} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - Caused by: java.lang.ClassNotFoundException: scala.$less$colon$less
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - ... 20 more
[2025-07-01T13:03:55.847+0000] {spark_submit.py:491} INFO - 
[2025-07-01T13:03:55.865+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Invoking stop() from shutdown hook
[2025-07-01T13:03:55.865+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-07-01T13:03:55.871+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkUI: Stopped Spark web UI at http://2b2f1149d0f1:4040
[2025-07-01T13:03:55.872+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-07-01T13:03:55.874+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-07-01T13:03:55.879+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-07-01T13:03:55.886+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO MemoryStore: MemoryStore cleared
[2025-07-01T13:03:55.887+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO BlockManager: BlockManager stopped
[2025-07-01T13:03:55.890+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-07-01T13:03:55.891+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-07-01T13:03:55.898+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO SparkContext: Successfully stopped SparkContext
[2025-07-01T13:03:55.898+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO ShutdownHookManager: Shutdown hook called
[2025-07-01T13:03:55.898+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO ShutdownHookManager: Deleting directory /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d
[2025-07-01T13:03:55.900+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO ShutdownHookManager: Deleting directory /tmp/spark-7473efdb-9873-4d16-b4d6-d5893047a13d/pyspark-f1a518b1-d5b0-4bbc-8a1e-322035108c6e
[2025-07-01T13:03:55.901+0000] {spark_submit.py:491} INFO - 25/07/01 13:03:55 INFO ShutdownHookManager: Deleting directory /tmp/spark-3d8d4cd2-b24e-4a07-825d-36d34832201f
[2025-07-01T13:03:55.919+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T13:03:55.926+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.
[2025-07-01T13:03:55.928+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T130330, end_date=20250701T130355
[2025-07-01T13:03:55.933+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task submit_grazing_detector_job (Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.; 181)
[2025-07-01T13:03:55.937+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T13:03:55.943+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T13:03:55.943+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-01T13:35:48.789+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T13:35:48.808+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T13:35:48.811+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T13:35:48.811+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-01T13:35:48.815+0000] {taskinstance.py:2330} INFO - Executing <Task(DockerOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T13:35:48.818+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpvzed9vka']
[2025-07-01T13:35:48.819+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=200) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T13:35:48.820+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask submit_grazing_detector_job
[2025-07-01T13:35:48.820+0000] {standard_task_runner.py:63} INFO - Started process 201 to run task
[2025-07-01T13:35:48.841+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host 207480bf3116
[2025-07-01T13:35:48.873+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T13:35:48.873+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T13:35:48.915+0000] {docker.py:359} INFO - Starting docker container from image bitnami/spark:3.5
[2025-07-01T13:35:48.920+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T13:35:48.920+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 275, in _raise_for_status
    response.raise_for_status()
  File "/home/airflow/.local/lib/python3.12/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http+docker://localhost/v1.48/containers/create

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 502, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 376, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 391, in _run_image_with_mounts
    self.container = self.cli.create_container(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/container.py", line 440, in create_container
    return self.create_container_from_config(config, name, platform)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/container.py", line 457, in create_container_from_config
    return self._result(res, True)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 281, in _result
    self._raise_for_status(response)
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 277, in _raise_for_status
    raise create_api_error_from_http_exception(e) from e
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/errors.py", line 39, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation) from e
docker.errors.APIError: 400 Client Error for http+docker://localhost/v1.48/containers/create: Bad Request ("invalid mount config for type "bind": invalid mount path: './airflow/spark_jobs' mount path must be absolute")
[2025-07-01T13:35:48.925+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T133548, end_date=20250701T133548
[2025-07-01T13:35:48.930+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task submit_grazing_detector_job (400 Client Error for http+docker://localhost/v1.48/containers/create: Bad Request ("invalid mount config for type "bind": invalid mount path: './airflow/spark_jobs' mount path must be absolute"); 201)
[2025-07-01T13:35:48.960+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T13:35:48.968+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T13:35:48.968+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-01T14:13:47.868+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T14:13:47.880+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T14:13:47.883+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [queued]>
[2025-07-01T14:13:47.883+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-01T14:13:47.887+0000] {taskinstance.py:2330} INFO - Executing <Task(DockerOperator): submit_grazing_detector_job> on 2024-12-31 15:00:00+00:00
[2025-07-01T14:13:47.889+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'scheduled__2024-12-31T15:00:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmp608qv2sb']
[2025-07-01T14:13:47.891+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask submit_grazing_detector_job
[2025-07-01T14:13:47.890+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=180) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T14:13:47.891+0000] {standard_task_runner.py:63} INFO - Started process 181 to run task
[2025-07-01T14:13:47.908+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job scheduled__2024-12-31T15:00:00+00:00 [running]> on host ff8c8c07f0c8
[2025-07-01T14:13:47.941+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-31T15:00:00+00:00'
[2025-07-01T14:13:47.941+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T14:13:47.977+0000] {docker.py:359} INFO - Starting docker container from image bitnami/spark:3.5
[2025-07-01T14:13:47.981+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T14:13:47.982+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 275, in _raise_for_status
    response.raise_for_status()
  File "/home/airflow/.local/lib/python3.12/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http+docker://localhost/v1.48/containers/create

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 502, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 376, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 391, in _run_image_with_mounts
    self.container = self.cli.create_container(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/container.py", line 440, in create_container
    return self.create_container_from_config(config, name, platform)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/container.py", line 457, in create_container_from_config
    return self._result(res, True)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 281, in _result
    self._raise_for_status(response)
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 277, in _raise_for_status
    raise create_api_error_from_http_exception(e) from e
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/errors.py", line 39, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation) from e
docker.errors.APIError: 400 Client Error for http+docker://localhost/v1.48/containers/create: Bad Request ("invalid mount config for type "bind": bind source path does not exist: /opt/project/airflow/spark_jobs")
[2025-07-01T14:13:47.987+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=scheduled__2024-12-31T15:00:00+00:00, execution_date=20241231T150000, start_date=20250701T141347, end_date=20250701T141347
[2025-07-01T14:13:47.991+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task submit_grazing_detector_job (400 Client Error for http+docker://localhost/v1.48/containers/create: Bad Request ("invalid mount config for type "bind": bind source path does not exist: /opt/project/airflow/spark_jobs"); 181)
[2025-07-01T14:13:48.032+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T14:13:48.039+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T14:13:48.040+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
